src/gates.jl

```diff
@@ -0,0 +1,115 @@
+# Gating variables (as an interface)
+abstract type AbstractGatingVariable end
+
+# TODO: hassteadystate(x::AbstractGatingVariable) = hasfield(typeof(x), :ss) ? !(isnothing(x.ss)) : false
+hasexponent(x::AbstractGatingVariable) = hasfield(typeof(x), :p) ? x.p !== one(Float64) : false
+getsymbol(x::AbstractGatingVariable) = only(states(x.sys))
+getequation(x::AbstractGatingVariable) = only(equations(x.sys))
+getdefaults(x::AbstractGatingVariable) = defaults(x.sys)
+subtype(x::AbstractGatingVariable) = typeof(x.sys)
+subtype(x::Type{G}) where G<:AbstractGatingVariable = fieldtype(G,:sys)
+
+struct Gate{S<:AbstractSystem} <: AbstractGatingVariable
+    sys::S # symbolic system
+    p::Float64 # optional exponent (defaults to 1)
+end
+
+abstract type AbstractGateModel end
+struct SteadyStateTau <: AbstractGateModel end
+struct AlphaBetaRates <: AbstractGateModel end
+
+function Gate(::Type{SteadyStateTau}, name::Symbol, ss::Num, tau::Num, p::Float64;
+    defaults::Dict=Dict(), sys_type::Type{S}=ODESystem) where S<:AbstractSystem
+    sym = only(@variables $name(t))
+
+    pars = Set{Num}()
+    pars_defaults = Dict()
+    for symbol in union(get_variables.([ss,tau])...)
+        if isparameter(symbol)
+            # Extra parameters are made unique to the gate ie. x => m₊x
+            par = renamespace(name,symbol)
```

Comment:
This will happen automatically with `getparameter(sys, :var)` overloading, so I don't think we need to (or want to) flatten the namespace here. We should also track all states. For instance
```julia
using Conductor, IfElse, ModelingToolkit

Vₘ = MembranePotential()
@parameters z

gate = Gate(AlphaBetaRates,
         αₘ = IfElse.ifelse(Vₘ == -40.0, 1.0, (0.1*(Vₘ + 40.0))/(1.0 - exp(-(Vₘ + 40.0)/10.0))),
         βₘ = 4.0*exp(-(Vₘ + z*65.0)/18.0),
         p = 3)

gate.sys.z # doesn't exist because we've forced a flattened name
gate.sys.m₊z # returns m₊m₊z
gate.sys.Vₘ # doesn't exist because we explicitly set states to only m(t)
```